common/gamelib/player.mjs

```diff
@@ -90,16 +96,18 @@ class Player {
   /**
    * Update the player data from the message.
    * @param {PlayerSyncMessage} msg - The message.
+   * @param {String} movingTime - Which moving time is to be updated. Empty string means don't update.
    * @return {Boolean} - Whether the update succeeds or not.
    */
-  updateFromMessage(msg) {
+  updateFromMessage(msg, movingTime='server') {
     if (msg.playerID !== this.playerID) {
       return false;
     }
     // TODO: check whether msg is in the correct format
     if (msg.mapCoord !== undefined) {
       if (this.mapCoord !== undefined && !msg.mapCoord.equalsTo(this.mapCoord)) {
-        this.lastMovingTime = Date.now();
+        if (movingTime === 'client' || movingTime === 'both') this.lastMovingTimeClient = Date.now();
```

Comment:
Add a check/assert to make sure it's one of the 3 possible values?

Reply:
added a warning